qiime2/sdk/tests/test_util.py

```diff
@@ -0,0 +1,37 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016-2018, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import unittest
+
+import qiime2
+import qiime2.sdk
+
+
+class TestUtil(unittest.TestCase):
+    def test_artifact_actions(self):
+        obs = qiime2.sdk.util.artifact_actions(None)
+        self.assertEqual(obs, [])
+
+        # For simplicity, we are gonna test the names of the plugin and
+        # the actions
+        obs = [(x.name, [yy.name for yy in y])
+               for x, y in qiime2.sdk.util.artifact_actions('SingleInt')]
+        exp = [('dummy-plugin', [
+            'Do stuff normally, but override this one step sometimes'])]
+        self.assertEqual(obs, exp)
+
+        obs = [(x.name, [yy.name for yy in y])
+               for x, y in qiime2.sdk.util.artifact_actions('Kennel[Cat]')]
+        self.assertEqual(obs, [])
+
+        with self.assertRaises(qiime2.sdk.util.UnknownTypeError):
+            qiime2.sdk.util.artifact_actions('ShouldBeEmpty')
+
```

Comment:
Let's add one more test to make sure that passing a concrete type like `IntSequence1` find actions that have unions like `IntSequence1 | IntSequence2`.